src/couch_mrview/src/couch_mrview_util.erl

```diff
@@ -786,26 +788,32 @@ changes_ekey_opts(_StartSeq, #mrargs{end_key=EKey,
     end.
 
 
+reduced_external_size(Tree) ->
+    case couch_btree:full_reduce(Tree) of
+        {ok, {_, _, Size}} -> Size;
+        % return null for versions of the reduce function without Size
+        {ok, {_, _}} -> null
```

Comment:
You're crossing streams a bit here. This returns null when there's no external size. But below your reduce function returns 0 when there's no external size. This means that with no write, we'll get null, but the first change will give us a number. We should pick one or the other.